src/resources/nfts/simplehash/utils.ts

```diff
@@ -193,3 +204,206 @@ export function simpleHashNFTToUniqueAsset(nft: SimpleHashNFT): UniqueAsset {
+        : undefined,
+      name:
+        uniqueTokenType === uniqueTokenTypes.ENS
+          ? ENS_COLLECTION_NAME
+          : collection.name ?? undefined,
+      simpleHashSpamScore: collection?.spam_score ?? undefined,
+      twitter: collection.twitter_username ?? undefined,
+    },
+    contract: {
+      address: nft.contract_address,
+      name: nft.contract.name ?? undefined,
+      schema_name: nft.contract.type,
+      symbol: nft.contract.symbol ?? undefined,
+    },
+    description: nft.description ?? undefined,
+    externalUrl: nft.external_url ?? undefined,
+    images: {
+      blurhash: nft.previews?.blurhash ?? undefined,
+      fullResPngUrl,
+      fullResUrl,
+      lowResPngUrl,
+      // mimeType only corresponds to fullResUrl
+      mimeType: nft.image_properties?.mime_type ?? undefined,
+    },
+    isSendable:
+      // can't send poaps because they're on gnosis
+      uniqueTokenType !== uniqueTokenTypes.POAP &&
+      (nft.contract.type === ERC721 || nft.contract.type === ERC1155),
+    lastSale:
+      nft.last_sale?.payment_token?.name &&
+      nft.last_sale?.payment_token?.symbol &&
+      !isNil(nft.last_sale?.unit_price)
+        ? {
+            paymentToken: {
+              address: nft.last_sale?.payment_token.address,
+              decimals: nft.last_sale?.payment_token.decimals,
+              name: nft.last_sale?.payment_token.name,
+              symbol: nft.last_sale?.payment_token.symbol,
+            },
+            value: nft.last_sale?.unit_price,
+          }
+        : undefined,
+    marketplaces: nft.collection.marketplace_pages
+      .filter(
+        (marketplace: SimpleHashMarketplace) =>
+          // filter out unsupported marketplaces
+          !!getInternalMarketplaceIdFromSimpleHashMarketplaceId(
+            marketplace.marketplace_id
+          )
+      )
+      .map(
+        (marketplace: SimpleHashMarketplace) =>
+          ({
+            collectionId: marketplace?.marketplace_collection_id,
+            collectionUrl: marketplace?.collection_url,
+            marketplaceId: getInternalMarketplaceIdFromSimpleHashMarketplaceId(
+              marketplace.marketplace_id
+            ),
+            name: marketplace?.marketplace_name,
+            nftUrl: marketplace?.nft_url,
+          } as NFTMarketplace)
+      ),
+    name: nft.name ?? undefined,
+    network,
+    predominantColor: nft.previews?.predominant_color ?? undefined,
+    tokenId: nft.token_id ?? undefined,
+    traits:
+      nft.extra_metadata?.attributes
+        // filter out traits that are missing key attributes
+        ?.filter(
+          (trait: SimpleHashTrait) =>
+            !isNil(trait.trait_type) && !isNil(trait.value)
+        )
+        ?.map(
+          (trait: SimpleHashTrait) =>
+            ({
+              displayType: trait?.display_type,
+              traitType: trait.trait_type,
+              value: trait.value,
+            } as NFTTrait)
+        ) ?? [],
```

Comment:
Nit: all these filter/maps, can we put those within the function body and not within the returned object? Just harder to read down here.